dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityStoreTest.java

```diff
@@ -308,11 +327,14 @@ void testQuery() {
     assertTrue(trackedEntitites.contains(trackedEntityE));
     assertTrue(trackedEntitites.contains(trackedEntityF));
     // Filter by program enrollment
-    params = new TrackedEntityQueryParams().setProgram(prA);
+    // TODO(tracker) This assertions fails, it returns only one TE. Couldn't figure out why yet, but
+    // I'm assuming this test class will be removed in the next iteration of DHIS2-17712. That's why
```

Comment:
[DHIS2-17712](https://dhis2.atlassian.net/browse/DHIS2-17712) is your issue `Expose one tracker enrollment service` will you remove the store? Or do you mean the issue @enricocolasante is working on?

If @enricocolasante is ok with it write `TODO(DHIS2-17714 tracker)` so we find the specific todo. Or do you Enrico know why this could be failing?

Reply:
I have no idea what is happening here.
It is up to you @muilpp to leave it like this and take care of it in your next PR or "assign" it to the TE task

Reply:
I meant this test should be removed in the ticket @enricocolasante is working on, along with the service class too. This assertion validates the TE's fetched filtering by program contain only the ones that are enrolled in that specific program, and this is already tested in the tracker service, so in my opinion, as soon as we remove this store and service, this test class will be no longer necessary.

@enricocolasante I don't want to assign this to the TE task, so if you think this will be still necessary I'll give it another shot and find what's wrong with it.